### PR TITLE
add disc selecting logic for powerkvm3.1.0

### PR DESCRIPTION
**Pull request by immarvin**

### CODE REVIEW

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		"mptsas"|"mpt2sas"|"mpt3sas") #*PROBABLY* not SAN, but SAS SAN is possible
+			if [ -z "$probablyfirstdirectdisk" ]; then probablyfirstdirectdisk=$disk; fi #remember first disk as a guess of medium resort
+			;;
+		*)
+			if [ -z "$firstdisk" ]; then firstdisk=$disk; fi #remember first disk as a guess of medium resort
+			;;
+		esac
+	done
+fi
+
```
Comment (zet809) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: For the 2 sections: 1 check /proc/paritions and 2 check /dev/sd*, why not merge them into 1 like this?
For every disk that can list:
    check the driver
         driver is ok, check diskname/devwwn, and write into temp files.
         Not ok, go the next partition device.
For the temp files, sort the output and choose the inst_disk.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#first take the disk with the MAX WWN numer to be the installdisk
+if [ -z "$instdisk" ]; then
+   disks=$(awk -F' '  '{print $4  }' /proc/partitions |grep -e "sd[a-z]$")
+   for disk in $disks; do
+      diskinfo=$(udevadm info --name $disk)
+      diskname=$(IFS= ;echo $diskinfo|grep '\<DEVNAME\>'|cut -d "=" -f2|tr A-Z a-z)
+      devtype=$(IFS= ;echo $diskinfo|grep '\<DEVTYPE\>'|cut -d "=" -f2|tr A-Z a-z)
+      devwwn=$(IFS= ;echo $diskinfo|grep '\<ID_WWN\>'|cut -d "=" -f2|tr A-Z a-z)
+      [  "$devtype" != "disk" ]  && continue
```
Comment (zet809) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: The devtype is useless if check the device driver first.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 (hunk 1 above)
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: thelogic here is from redhat/sles  prescript, I would like to keep it here as the backup when the newly added logic failed to determine the disk to partition. I am wondering whether we should delete or keep the logic here.  I suggest that we keep it for this release and refine it and make it consistent for redhat,sles and ubuntu in next release.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 (hunk 2 above)
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: yes

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 [hunk 3]
```diff
@@ -0,0 +1,58 @@
+#find first available block device (sda or vda likely)
+#TODO: pick a likely non-SAN target if possible
+
+shopt -s nullglob
+
+#first take the disk with the MAX WWN numer to be the installdisk
+if [ -z "$instdisk" ]; then
+   disks=$(awk -F' '  '{print $4  }' /proc/partitions |grep -e "sd[a-z]$")
```
Comment (zet809) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: I think it shall sd[a-z]+  here for the scenario that there is a lot of disks.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 (hunk 2 above)
Comment (zet809) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: I mean is it possible the it is not a disk in $disks?
We don't need to check the devtype here, right?

Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: this piece of code does not focus on the device driver, it just check the WWN. The guess on device driver is done in the logic below if the newly added logic cannot determine the disc.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 (hunk 3 above)
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: the available disk should be "sda,sdb...", I think "sd[a-z]" is ok. "sda1,sda2" are partitions, which cannot be used to create partitions.

File: xCAT-server/share/xcat/install/scripts/pre.pkvm3 (hunk 1 above)
Comment (zet809) [inline xCAT-server/share/xcat/install/scripts/pre.pkvm3]: OK

**Comment by zet809** [comment]
Look good to me, I agree to merge.

**Pull request merged**